src/internal/streamstore/streamstore.go

```diff
@@ -18,24 +18,102 @@ import (
 	"github.com/alcionai/corso/src/pkg/path"
 )
 
+// ---------------------------------------------------------------------------
+// controller
+// ---------------------------------------------------------------------------
+
+var _ Streamer = &storeStreamer{}
+
+type storeStreamer struct {
+	kw      *kopia.Wrapper
+	tenant  string
+	service path.ServiceType
+	dbcs    []data.BackupCollection
+}
+
+// NewStreamer creates a new streamstore Streamer for stream writing metadata files
+// to the store.
+func NewStreamer(
+	kw *kopia.Wrapper,
+	tenant string,
+	service path.ServiceType,
+) *storeStreamer {
+	return &storeStreamer{
+		kw:      kw,
+		tenant:  tenant,
+		service: service,
+	}
+}
+
+func (ss *storeStreamer) Collect(ctx context.Context, col Collectable) error {
```

Comment:
This needs a comment that the data is eagerly serialized

Reply:
done